pkg/service/json.go

```diff
@@ -160,11 +160,14 @@ func ParseJSONBody(body io.Reader, obj interface{}) error {
 func ValidateStruct(obj interface{}) error {
 	err := validate.Struct(obj)
 	if err != nil {
-		switch err := err.(type) {
-		case *validator.InvalidValidationError:
+		var invalidValidationErr *validator.InvalidValidationError
+		if errors.As(err, &invalidValidationErr) {
 			return NewInvalidRequestError("Invalid request body")
```

Comment:
Same comment as above: Can you log err above this return for easier debugging? Something like `log.Error().Err(err).Msgf("json: invalid request body: ValidateStruct")`